clients/unshaded/src/main/java/tachyon/client/file/options/CreateFileOptions.java

```diff
@@ -0,0 +1,195 @@
+  /**
+   * @return the TTL (time to live) value; it identifies duration (in milliseconds) the created file
+   *         should be kept around before it is automatically deleted
+   */
+  public long getTTL() {
+    return mTTL;
+  }
+
+  /**
+   * @return the under storage type
+   */
+  public UnderStorageType getUnderStorageType() {
+    return mUnderStorageType;
+  }
+
+  /**
+   * @return whether or not the recursive flag is set
+   */
+  public boolean isRecursive() {
+    return mRecursive;
+  }
+
+  /**
+   * @param blockSizeBytes the block size to use
+   * @return the builder
+   */
+  public CreateFileOptions setBlockSizeBytes(long blockSizeBytes) {
+    mBlockSizeBytes = blockSizeBytes;
+    return this;
+  }
+
+  /**
+   * @param locationPolicy the location policy to use
+   * @return the builder
+   */
+  public CreateFileOptions setHostname(FileWriteLocationPolicy locationPolicy) {
+    mLocationPolicy = locationPolicy;
+    return this;
+  }
+
+  /**
+   * @param recursive whether or not to recursively create the file's parents
+   * @return the updated object
+   */
+  public CreateFileOptions setRecurisve(boolean recursive) {
+    mRecursive = recursive;
+    return this;
+  }
+
+  /**
+   * @param ttl the TTL (time to live) value to use; it identifies duration (in milliseconds) the
+   *        created file should be kept around before it is automatically deleted, no matter whether
+   *        the file is pinned
+   * @return the builder
+   */
+  public CreateFileOptions setTTL(long ttl) {
+    mTTL = ttl;
+    return this;
+  }
+
+  /**
+   * @param writeType the {@link tachyon.client.WriteType} to use for this operation. This will
+   *        override both the TachyonStorageType and UnderStorageType.
+   * @return the builder
+   */
+  public CreateFileOptions setWriteType(WriteType writeType) {
+    mTachyonStorageType = writeType.getTachyonStorageType();
+    mUnderStorageType = writeType.getUnderStorageType();
+    return this;
+  }
+
+  public OutStreamOptions toOutStreamOptions() {
+    OutStreamOptions.Builder builder = new OutStreamOptions.Builder();
+    builder.setBlockSizeBytes(mBlockSizeBytes).setLocationPolicy(mLocationPolicy)
+        .setTachyonStorageType(mTachyonStorageType).setTTL(mTTL)
+        .setUnderStorageType(mUnderStorageType);
+    return builder.build();
+  }
+
+  /**
+   * @return the name : value pairs for all the fields
+   */
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder("CreateFileOptions(");
+    sb.append(super.toString()).append(", BlockSizeBytes: ").append(mBlockSizeBytes);
+    sb.append(", Location Policy: ").append(mLocationPolicy);
+    sb.append(", TachyonStorageType: ").append(mTachyonStorageType.toString());
+    sb.append(", UnderStorageType: ").append(mUnderStorageType.toString());
+    sb.append(", TTL: ").append(mTTL);
+    sb.append(")");
+    return sb.toString();
+  }
+
+  /**
+   * @return Thrift representation of the options
+   */
```

Comment:
add a TODO to rename to CreateFileTOptions? (if we are renaming MkdirTOptions to CreateDirectoryTOptions)